src/Utilities.UnitTests/ToolLocationHelper_Tests.cs

```diff
@@ -2000,34 +1996,20 @@ public void GetPathToReferenceAssembliesDefaultLocation99()
             directories.Count.ShouldBe(0); // "Expected the method to return no paths."
         }
 
-#if FEATURE_SPECIAL_FOLDERS
         /// <summary>
-        /// Make sure we choose the correct path for program files based on the environment variables
+        /// Make sure we choose the correct path for program files based on the operating system
         /// </summary>
         [Fact]
-#else
-        [Fact(Skip = "Special folders not supported")]
-#endif
         [PlatformSpecific(TestPlatforms.Windows)]
         public void TestGenerateProgramFiles32()
         {
-            string programFilesX86Original = Environment.GetEnvironmentVariable("ProgramFiles(x86)");
-            string programFiles = Environment.GetEnvironmentVariable("ProgramFiles");
-            try
-            {
-                Environment.SetEnvironmentVariable("ProgramFiles(x86)", null);
-                string result = FrameworkLocationHelper.GenerateProgramFiles32();
-                result.ShouldBe(programFiles, StringCompareShould.IgnoreCase); // "Expected to use program files but used program files x86"
+            Environment.SpecialFolder folder = Environment.Is64BitOperatingSystem ? Environment.SpecialFolder.ProgramFilesX86 : Environment.SpecialFolder.ProgramFiles;
```

Comment:
The was definitely some weird behavior. The return value of ```GetEnvironmentVariable("ProgramFiles")``` seems to depend on whether the process itself is 32-bit or 64-bit, so on a 64-bit OS ```ProgramFiles(x86)``` would always be the expected value even though the test tried to simulate a 32-bit OS. Setting the environment variables also made no difference so the return values. So when this was enabled on netcore, it would always fail even though ```ProgramFiles(x86)``` correctly returned.